### DISCUSSION TITLE
If I have a string that's already formatted, can I print it with rich?

### DISCUSSION THREAD

**Discussion by dbieber** [Textualize/rich#3253] [Q&A] [answered]
If I have a string that's already formatted, can I print it with rich?

i.e. suppose I capture from a subprocess bytes like:
```python
captured_bytes = b'\x1b[1;34mThis is blue bold text\x1b[0m'
```
If I naively print it with rich, it will print with the ANSI code visible.

<img width="652" alt="image" src="https://github.com/Textualize/rich/assets/892765/b40692d6-06df-40ea-bfd3-d03f19fdbbd1">

Is there a way to transform captured_bytes into a renderable that will render properly (i.e. as blue bold text)?

**Comment by TomJGooding** [comment, marked as answer]
You could use [`Text.from_ansi`](https://rich.readthedocs.io/en/stable/reference/text.html?highlight=ansi#rich.text.Text.from_ansi) to convert the strings into something that Rich can print.

**Reply by dbieber** [reply]
Was just about to post that myself; thank you!
